File: challenges/3-tipos-de-triangulo/javascript/codinghobbit/desafio3.js
function isTriangle(a, b, c) {
  return !(a >= b + c);
}

function triangleType(a, b, c) {
  return a === b && b === c
    ? "TRIANGULO EQUILATERO"
    : a === b || b === c || c === a
    ? "TRIANGULO ISOSCELES"
    : "TRIANGULO ESCALENO";
}

function triangleAngle(a, b, c) {
  return Math.pow(a, 2) < Math.pow(b, 2) + Math.pow(c, 2)
    ? "TRIANGULO ACUTANGULO"
    : Math.pow(a, 2) > Math.pow(b, 2) + Math.pow(c, 2)
    ? "TRIANGULO OBTUSANGULO"
    : "TRIANGULO RETANGULO";
}

const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

readline.question("Insira a medida dos três lados do triângulo: ", (edges) => {
  const edgesArr = edges.split(" ");
  const edgesArrSorted = edgesArr.sort((a, b) => a - b);
  const [A, B, C] = edgesArrSorted.reverse();
  if (isTriangle(A, B, C)) {
    console.log(triangleAngle(A, B, C));
    console.log(triangleType(A, B, C));
  } else console.log("NAO FORMA TRIANGULO");
  readline.close();
});

File: challenges/3-tipos-de-triangulo/python/camiscf/desafio3.py
#tipos de triangulo

a, b, c = input().split()
a = float(a)
b = float(b)
c = float(c)

# ordem descrecente
if a < b:
    a, b = b, a
if b < c:
    b, c = c, b
if a < b:
    a, b = b, a

if a >= b + c:
    print("NAO FORMA TRIANGULO")
else:
    if a**2 == b**2 + c**2:
        print("TRIANGULO RETANGULO")
    if a**2 > b**2 + c**2:
        print("TRIANGULO OBTUSANGULO")
    if a**2 < b**2 + c**2:
        print("TRIANGULO ACUTANGULO")
    if a == b == c:
        print("TRIANGULO EQUILATERO")
    if (a == b != c) or (a == c != b) or (b == c != a):
        print("TRIANGULO ISOSCELES")

